src/runmode-pfring.c

```diff
@@ -323,24 +323,28 @@ static void *ParsePfringConfig(const char *iface)
             if (unlikely(pfconf->bpf_filter == NULL)) {
                 SCLogError("Can't allocate BPF filter string");
             } else {
-                SCLogDebug("Going to use command-line provided bpf filter %s",
-                           pfconf->bpf_filter);
+                SCLogConfig("%s: using to use command-line provided bpf filter %s", pfconf->iface,
+                        pfconf->bpf_filter);
             }
         }
-    } else {
+    } else if (pfconf->bpf_filter == NULL) {
         if (ConfGetChildValueWithDefault(if_root, if_default, "bpf-filter", &bpf_filter) == 1) {
             if (strlen(bpf_filter) > 0) {
                 pfconf->bpf_filter = SCStrdup(bpf_filter);
                 if (unlikely(pfconf->bpf_filter == NULL)) {
                     SCLogError("Can't allocate BPF filter string");
                 } else {
-                    SCLogDebug("Going to use bpf filter %s",
-                               pfconf->bpf_filter);
+                    SCLogConfig("%s: using file provided bpf filter %s", pfconf->iface,
+                            pfconf->bpf_filter);
                 }
             }
         }
     }
 
+    if (pfconf->bpf_filter != NULL && EngineModeIsIPS()) {
```

Comment:
we don't support IPS for pfring

Reply:
removed the bpf_filter test from the condition but left the IPS test to stop user from running Suricata with PF_RING and IPS mode.